### PR TITLE
KAFKA-4849: Bug in KafkaStreams documentation

### PR DESCRIPTION
**Pull request by mjsax**
- removed ZK config from Streams cods
 - removed deprecated ZK config parameter from configs table

### CODE REVIEW

**Comment by mjsax** [comment]
call for review @ewencp @guozhangwang @enothereska @dguy

File: streams/src/main/java/org/apache/kafka/streams/kstream/Transformer.java
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/kstream/Transformer.java:33]: This is just an unrelated typo fix. Discovered the typo by chance.

**Comment by mjsax** [comment]
Not sure how we can get this into `0.10.2` to update the web page...? Anyone?

File: clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java
Comment (dguy) [inline clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java:998]: You probably need to do the same in `toEnrichedRst()` and `toRst()`

**Comment by mjsax** [comment]
Updated.

**Review by dguy — APPROVED** [review]
LGTM

**Comment by guozhangwang** [comment]
LGTM. @ijuma do you want to take another look?

File: clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java
Comment (ijuma) [inline clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java:1040]: Are we just skipping deprecated configs? The approach in the following PR from @cmccabe seems better:

https://github.com/apache/kafka/pull/2456/files

Comment (mjsax) [reply clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java:1040]: I did not know about the other Jira. I am fine either way.

Comment (mjsax) [reply clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java:1040]: @ijuma @cmccabe @guozhangwang Any news about this? What's the status of #2456 ?

Comment (guozhangwang) [reply clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java:1040]: I think marking it as deprecated in html would be better than skipping it. So if #2456 is definitely going in we can close this PR then.

Comment (mjsax) [reply clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java:1040]: Still need to mark the ZK config as deprecated -- ie, need to update the call to `define` within `StreamsConfig` -- or will #2456 cover this part, too. \cc @cmccabe

Comment (guozhangwang) [reply clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java:1040]: Actually I think #2456 does not update the printed html for marking these as deprecated, so we still need to rebase + merge this PR after that one is merged.

**Comment by guozhangwang** [comment]
Is this still valid?

**Comment by mjsax** [comment]
Yes it is. But it is still blocked by #2456 -- seems that @cmccabe does not find time to work on it.

**Comment by guozhangwang** [comment]
@mjsax would you like to start a discussion on removing some deprecated interfaces and configs in the coming release (1.2.0) including this PR?

**Comment by mjsax** [comment]
Sure. Should we do this right away? Or should we wait after 1.1 is released?

**Comment by guozhangwang** [comment]
@mjsax let's wait after 1.1.0 is out the door :)

**Comment by mjsax** [comment]
We recently removed deprecated ZK config completely. Thus, this PR is void now. Closing.

**Pull request closed without merging**
